lib/productions/helpers.js

```diff
@@ -242,3 +242,43 @@ export function findLastIndex(array, predicate) {
   }
   return array.length - index - 1;
 }
+
+/**
+ * Returns a proxy that auto-assign `parent` field.
+ * @template T
+ * @param {T} data
+ * @param {*} parent this is `data` by default
+ * @return {T}
+ */
+export function autoParenter(data, parent) {
+  if (!parent) {
+    parent = data;
+  }
+  if (!data || typeof data !== "object") {
```

Comment:
Similarly here... explaining why this is not "exceptional" would be good (I guess this is parent root or something).